File: typescript/test/doubles/TestOrderRepository.ts
import Order from '../../src/domain/Order';
import OrderRepository from '../../src/repository/OrderRepository';

class TestOrderRepository implements OrderRepository {
  private insertedOrder: Order = null;
  private orders: Order[] = [];

  public getSavedOrder(): Order {
      return this.insertedOrder;
  }

  public save(order: Order): void {
      this.insertedOrder = order;
  }

  public getById(orderId: number): Order {
      return this.orders.find(o => o.getId() == orderId);
  }

  public addOrder(order: Order): void {
      this.orders.push(order);
  }
}

export default TestOrderRepository;

File: python/test/doubles/test_order_repository.py
from ...src.repository.order_repository import OrderRepository


class TestOrderRepository(OrderRepository):
    def __init__(self):
        self.inserted_order = None
        self.orders = []

    def save(self, order):
        self.inserted_order = order

    def get_by_id(self, order_id):
        return next((o for o in self.orders if o.id == order_id), None)

    def add_order(self, order):
        self.orders.append(order)
